relecov_tools/upload_database.py

```diff
@@ -266,15 +285,19 @@ def update_database(self, field_values, post_url):
                     f"[blue] sending request for sample {chunk['sequencing_sample_id']}"
                 )
                 req_sample = chunk["sequencing_sample_id"]
+
             self.logsum.feed_key(sample=req_sample)
+
             result = self.platform_rest_api.post_request(
                 json.dumps(chunk),
                 {"user": self.user, "pass": self.passwd},
                 post_url,
             )
+
             if "ERROR" in result:
                 if result["ERROR"] == "Server not available":
                     # retry to connect to server
+                    i = 0
```

Comment:
This does not generate any change in the code as `i` is already declared within the first step of the loop

Reply:
no, it does not, but again is a recommendation of pylance, that recommends having all variables declared. I can remove it if you think it's better

Reply:
I think that it would make more sense to remove the later conditional section from out of the loop so it is tested at the end of it with an else clause, like this:
```suggestion
for i in range(10):
      ...
else:
      if "ERROR" in result:
            logtxt = f"Unable to sent the request to {post_url}"
            self.logsum.add_error(entry=logtxt, sample=req_sample)
            stderr.print(f"[red]{logtxt}")
            continue
```


that way you dont declare `i` for nothing

Reply:
but the for is for retrying the connection, you only want to do it if error in result:
```
            result = self.platform_rest_api.post_request(
                json.dumps(chunk),
                {"user": self.user, "pass": self.passwd},
                post_url,
            )

            if "ERROR" in result:
                if result["ERROR"] == "Server not available":
                    # retry to connect to server
                    i = 0
                    for i in range(10):
                        # wait 5 sec before resending the request
                        time.sleep(5)
                        result = self.platform_rest_api.post_request(
                            json.dumps(chunk),
                            {"user": self.user, "pass": self.passwd},
                            self.platform_settings[post_url],
                        )
                        if "ERROR" not in result:
                            break
                    if i == 9 and "ERROR" in result:
                        logtxt = f"Unable to sent the request to {post_url}"
                        self.logsum.add_error(entry=logtxt, sample=req_sample)
                        stderr.print(f"[red]{logtxt}")
                        continue
```

In any case i've removed the i=0 as you say it's not needed with range(10), i don't know why pylance complained about it...it's not complaining now so